server/plugin/virtual_agent_test.go

```diff
@@ -357,40 +359,65 @@ func Test_CreateMessageAttachment(t *testing.T) {
 	defer monkey.UnpatchAll()
 
 	for _, testCase := range []struct {
-		description      string
-		fileID           string
-		response         *MessageAttachment
-		getFileInfoError *model.AppError
-		marshalError     error
-		encryptError     error
-		expectedError    string
+		description   string
+		userID        string
+		response      *MessageAttachment
+		setupAPI      func(api *plugintest.API)
+		marshalError  error
+		encryptError  error
+		expectedError string
 	}{
 		{
 			description: "CreateMessageAttachment returns a valid attachment",
-			fileID:      "mockFileID",
+			userID:      testutils.GetID(),
 			response: &MessageAttachment{
 				URL:         "mockSiteURL" + p.GetPluginURLPath() + "/file/" + encode([]byte{}),
 				ContentType: "mockMimeType",
 				FileName:    "mockName",
 			},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetFileInfo", mock.AnythingOfType("string")).Return(testutils.GetFile(false), nil)
+			},
 		},
 		{
 			description: "CreateMessageAttachment returns an error while getting file info",
-			fileID:      "mockFileID",
-			getFileInfoError: &model.AppError{
-				Message: "error in getting the file info",
+			userID:      testutils.GetID(),
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetFileInfo", mock.AnythingOfType("string")).Return(nil, testutils.GetAppError("error in getting the file info"))
 			},
 			expectedError: "error getting the file info. Error: error in getting the file info",
 		},
 		{
-			description:   "CreateMessageAttachment returns an error while marshaling file",
-			fileID:        "mockFileID",
+			description: "CreateMessageAttachment returns an error because the file is a deleted file",
```

Comment:
I think this is better 
```suggestion
			description: "CreateMessageAttachment returns an error because the file is already deleted",
```